runtime/lua/vim/filetype/detect.lua

```diff
@@ -572,10 +642,15 @@ function M.m(bufnr)
   end
 end
 
-function M.m4(path)
-  path = path:lower()
-  if not path:find('html%.m4$') and not path:find('fvwm2rc') then
-    return 'm4'
+local function m4(contents)
+  for _, line in ipairs(contents) do
+    if matchregex(line, '^\\%s*dnl\\>') then
```

Comment:
```suggestion
    if matchregex(line, [[^\s*dnl\\>]]) then
```
(not a Lua pattern ;))

Should fix the ASAN failure (that was a memory leak for the error, which for some reason doesn't show up in the tests).

If you haven't already otherwise, maybe it would make sense to use `[[...]]` strings for Vim regexes consistently to avoid the danger of wrong escaping (and simplify copy&pasting)?

Reply:
> If you haven't already otherwise, maybe it would make sense to use [[...]] strings for Vim regexes consistently

Yes, I think this was done every time `matchregex` was used :+1: